### ISSUE TITLE
script blocked

### ISSUE THREAD

**Issue by xycloud** [OptimalBits/bull#885]
bull version: 3.3.10
code: 
```
let Q = require('bull');

const rq = new Q('test', {redis: {
        port: 6379,
        host: 'xxx.x.x.x',
        password: 'xxx'
}});
```

when I run this code, its blocked, seems step into a event loop
why? we cannot  dynamically add job in different module cause the block

**Comment by manast** [comment]
Please add a complete code snippet that reproduce the problem.

**Comment by xycloud** [comment]
that is the complete code, and cause I cannot public redis information, so replaced with some meaningless character

**Comment by manast** [comment]
that code you provided just instantiates a queue, you have to provide the code that reproduces an issue.

**Comment by xycloud** [comment]
producer.js:

```
let Q = require('bull');

const rq = new Q('test', {redis: {
        port: 6379,
        host: 'xxx',
        password: 'xxx'
    }});


rq.add('test', {hello: 'bull'});
```

consumer.js:

```
let Q = require('bull');

const rq = new Q('test', {redis: {
        port: 6379,
        host: 'xxx',
        password: 'xxx'
    }});


rq.process(function(job, done) {
    console.log(job.data);
    done();
});

```


problem:
this producer.js module is blocked

**Comment by xycloud** [comment]
one more thing, cannot we add cron job dynamically? I run consumer.js first, and add cron job by producer.js later, then we got nothing output

**Comment by manast** [comment]
What do you mean with "is blocked" ?

**Comment by xycloud** [comment]
consumer.js should exit after run, but it won't

**Comment by manast** [comment]
try this:
```js
rq.add('test', {hello: 'bull'}).then(function(){
  rq.close();
});
```

**Comment by xycloud** [comment]
yes, thanks @manast , how about below question:

```
one more thing, cannot we add cron job with same cron config? seems bull:test:repeat is a zset structure for cron config
```

**Comment by manast** [comment]
Can you rephrase the question please?

**Comment by manast** [comment]
there is a feature request for updating existing repeatable jobs, please check the issues for workaround and information.

**Comment by xycloud** [comment]
I made a pr, may be not the best solution: https://github.com/OptimalBits/bull/pull/886

could you share the workaround?
